app/src/main/java/com/edulectronics/tinycircuit/Controllers/WireController.java

```diff
@@ -118,4 +134,47 @@ public Line getVerticalWire(Point a, Point b) {
             return null;
         return new Line(new Point(a.x, a.y), new Point(a.x, b.y - (Math.abs(a.y - b.y) % 150)));
     }
+
+    // First go half a cell up/down/left/right, ddepending on where the connectionpoint is
+    // and where the endpoint is. Otherwise the next lines will be drawn right through other
+    // components.
+    public Line getEndPointLine(Point startPoint, Point endPoint, ConnectionPointOrientation orientation) {
+        switch (orientation) {
+            case Left:
+            case Right:
+                if (startPoint.y < endPoint.y) {
+                    // Go up half a cell
+                    return new Line(startPoint,
+                            new Point(startPoint.x, startPoint.y + (int) (0.5 * cellHeight)));
+                } else {
+                    // Go down half a cell
+                    return new Line(startPoint,
+                            new Point(startPoint.x, startPoint.y - (int) (0.5 * cellHeight)));
+                }
+            case Bottom:
+            case Top:
+                if (startPoint.x > endPoint.x) {
+                    // Go left half a cell
+                    return new Line(startPoint,
+                            new Point(startPoint.x - (int) (0.5 * cellWidth), startPoint.y));
+                } else {
+                    // Go right half a cell
+                    return new Line(startPoint,
+                            new Point(startPoint.x + (int) (0.5 * cellWidth), startPoint.y));
+                }
+            default:
+                throw new IllegalArgumentException();
+        }
+    }
+
+    private void createConnection(Component component, MotionEvent event) {
+        Connector connector = new Connector();
+        ConnectionPointOrientation cpoSecond = clickedArea((int) event.getX(), (int) event.getY());
+        connector.connect(getConnectionPoint(first, cpoFirst), getConnectionPoint(component, cpoSecond));
```

Comment:
Connector.connect is trouwens static dus je hoeft geen connector object te maken